src/AndcultureCode.CSharp.Sitefinity.Core/Services/ODataServices.cs

```diff
@@ -0,0 +1,393 @@
+                return result;
+            }
+
+            result.ResultObject = JsonConvert.DeserializeObject<ODataServiceGetResponse<TContent>>(restResponse.Content).Value;
+
+            return result;
+        }
+
+        /// <summary>
+        /// Gets the number of the existing, published Sitefinity items for data type specified by this service
+        /// </summary>
+        /// <returns>The response of the request wrapped in a RestResponseResult where ResultObject is the total number of published records</returns>
+        public RestResponseResult<int> GetCount()
+        {
+            var requestUrl = $"{Settings.BaseUrl}{EndpointUrl}/$count";
+
+            var restResponse = ExecuteSitefinityRequest(Method.GET, requestUrl, null);
+
+            var result = new RestResponseResult<int>(HttpStatusCode.OK, restResponse);
+
+            if (result.WasUnexpectedStatusCode)
+            {
+                result.AddUnexpectedStatusCodeError(HttpStatusCode.OK, restResponse, "getting the total count of published");
+
+                return result;
+            }
+
+            var content = restResponse.Content.Trim(new char[] { '\uFEFF' }); // Removing the leading Byte Order Mark (BOM) from content.
+
+            result.ResultObject = Convert.ToInt32(content);
+
+            return result;
+        }
+
+        /// <summary>
+        /// Gets an existing item.
+        /// </summary>
+        /// <param name="id">The Id of the content.</param>
+        /// <returns>The response of the request wrapped in a RestResponseResult where ResultObject is the returned object</returns>
+        public RestResponseResult<TContent> GetItem(Guid id)
+        {
+            var requestUrl = $"{Settings.BaseUrl}{EndpointUrl}({id})";
+
+            var restResponse = ExecuteSitefinityRequest(Method.GET, requestUrl, null);
+
+            var result = new RestResponseResult<TContent>(HttpStatusCode.OK, restResponse);
+
+            if (result.WasUnexpectedStatusCode)
+            {
+                result.AddUnexpectedStatusCodeError(HttpStatusCode.OK, restResponse, "getting item");
+
+                return result;
+            }
+
+            result.ResultObject = JsonConvert.DeserializeObject<TContent>(restResponse.Content);
+
+            return result;
+        }
+
+        /// <summary>
+        /// Gets related items based on the relationship name.
+        /// NOTE: Use this for multi-instance relationships
+        /// </summary>
+        /// <param name="id">The Id of the content.</param>
+        /// <param name="string">The name of the relationship.</param>
+        /// <returns>The response of the request wrapped in a RestResponseResult where ResultObject is the returned related object list</returns>
+        public RestResponseResult<List<TRelatedContent>> GetRelatedMany<TRelatedContent>(Guid id, string relationshipName) where TRelatedContent : SitefinityContentDto
```

Comment:
@HeyKos  made a valid point in the previous project specific PR regarding naming here.  We could consider renaming this to something else like `GetRelatedManyRecords` and therefore the other method could be  something like `GetRelatedSingleRecord`.  Or maybe just `GetRelatedRecords` and `GetRelatedRecord` but I'm not too keen on that.  I'm okay with it also staying the way it is.  Just throwing more ideas out there if it adds any clarity to the naming.

Reply:
Personally I actually like `GetRelatedRecords` and `GetRelatedRecord` because I believe that's conventional in core C# classes -- see `Type.GetConstructor` and `Type.GetConstructors` or `Enum.GetName` and `Enum.GetNames`.

Reply:
Sounds good to me then!